js/components/dataset/resource/file-form.vue

```diff
@@ -128,18 +134,17 @@ export default {
             if (typeof this.dataset !== 'undefined') {
                 params = {dataset: this.dataset.id};
             }
-            if (this.is_community) {
-                endpoint += 'community_';
-                params.community = this.resource.id;
-            } else {
-                endpoint += 'dataset_';
-            }
             if (this.resource.id) {
-                endpoint += 'resource';
-                params.rid = this.resource.id;
+                if (this.is_community) {
+                    params.community = this.resource.id;
+                } else {
+                    params.rid = this.resource.id;
+                }
             } else {
-                endpoint += 'resources';
+                endpoint += 'new_';
             }
+            endpoint += this.is_community ? 'community_' : 'dataset_';
```

Comment:
Seems very error prone. Could we rather build a string like this?

```js
const route_new = this.resource.id ? '' : 'new_';
const route_namespace = this.is_community ? 'community_' : 'dataset_'; 

const endpoint = `upload_${route_new}${route_namespace}resource`;
```

Would make stuff more explicit, and probably simplify the above logic.

Reply:
Yes we can (and I will), but that's only working for the string building part, not the parameters.